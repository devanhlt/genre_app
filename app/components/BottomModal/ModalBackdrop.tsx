import React, { useMemo } from "react"

import { BottomSheetBackdropProps } from "@gorhom/bottom-sheet"
import Animated, { Extrapolate, interpolate, useAnimatedStyle } from "react-native-reanimated"

export const ModalBackdrop = ({ style }: BottomSheetBackdropProps) => {
  // animated variables
  const containerAnimatedStyle = useAnimatedStyle(() => ({
    opacity: interpolate(1, [0, 1], [0, 1], Extrapolate.CLAMP),
  }))

  // styles
  const containerStyle = useMemo(
    () => [
      style,
      {
        backgroundColor: "rgba(0,0,0,0.3)",
      },
      containerAnimatedStyle,
    ],
    [style, containerAnimatedStyle],
  )

  return <Animated.View style={containerStyle} />
}
