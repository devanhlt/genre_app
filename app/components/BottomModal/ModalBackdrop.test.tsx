/* eslint-disable react-native/no-inline-styles */
/* eslint-disable react-native/no-color-literals */
import React from "react"

import renderer from "react-test-renderer"
import { ModalBackdrop } from "./ModalBackdrop"

describe("<ModalBackdrop />", () => {
  it("renders correctly ", () => {
    const tree = renderer.create(
      <ModalBackdrop
        style={{ backgroundColor: "red" }}
        animatedIndex={{
          value: 0,
        }}
        animatedPosition={{
          value: 0,
        }}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
