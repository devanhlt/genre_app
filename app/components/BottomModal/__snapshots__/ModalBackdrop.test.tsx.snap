// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBackdrop /> renders correctly  1`] = `
<View
  style={
    [
      {
        "backgroundColor": "red",
      },
      {
        "backgroundColor": "rgba(0,0,0,0.3)",
      },
      {
        "opacity": undefined,
      },
    ]
  }
/>
`;
